predictive_tracking_V2.py

```diff
@@ -0,0 +1,321 @@
+        Estimate, = plt.plot(estimated_x[3:], estimated_y[3:], 'r-', label='Line 2')
+        plt.xlim(0, 500)
+        plt.ylim(360, 0)
+        plt.title('Real vs Estimated Position')
+        plt.xlabel('Pixel x')
+        plt.ylabel('Pixel y')   
+        plt.legend([Real, Estimate], ['Real', 'Estimate'])
+        plt.grid(True)
+        
+        plt.savefig('results.png')    
+        
+        
+        
+    def plot_prediction_results(self, estimated_x, estimated_y, predicted_x, predicted_y):
+        # plot Animals real location and estimated location
+        Estimate, = plt.plot(estimated_x[5:], estimated_y[5:], 'b-', label='Line 1')
+        Prediction, = plt.plot(predicted_x[5:], predicted_y[5:], 'r-', label='Line 2')
+        plt.xlim(0, 500)
+        plt.ylim(360, 0)
+        plt.title('Real vs Estimated Position')
+        plt.xlabel('Pixel x')
+        plt.ylabel('Pixel y')   
+        plt.legend([Estimate, Prediction], ['Estimate', 'Prediction'])
+        plt.grid(True)
+        
+        plt.savefig('Prediction.png')         
+    
+    
+    
+    def predict_location(self, frame, center, angle, velocity):
+        # Return and plot the future predicted location 
+        offset = np.array([0,0])
+        distance = np.array([0,0])
+        
+        # determine offset distance
+        time = 4
+        distance = velocity*time
+        #print('distance: ', distance)
+        length = np.linalg.norm(distance)
+        #print('length: ', length)
+        
+        # Find location coordinate
+        offset[0] = int(length*(math.cos(math.radians(angle))))
+        offset[1] = int(length*(math.sin(math.radians(angle))))    
+        location = center - offset
+        #print('Future location: ', location)
+        
+        # Plot line between current center and projected point
+        cv2.circle(frame, tuple(location), 5, self.Red, thickness=-1)   
+        cv2.circle(frame, tuple(location), 10, self.Red, thickness=2) 
+        
+        return location
+    
+    
+    
+    def plot_position(self, frame, center, estimated_center, predicted_center):
+        # Initialise text function
+        font = cv2.FONT_HERSHEY_SIMPLEX 
+        fontsize = 0.5
+        font_thickness = 1
+        
+        # Plot real, estimated and predicted position of animal
+        #cv2.circle(frame, tuple(center), 5, self.Red, -1) 
+        cv2.circle(frame, tuple(estimated_center), 5, self.Blue, -1) 
+        #cv2.circle(frame, tuple(predicted_center), 5, self.Green, -1) 
+        
+        #cv2.putText(frame,'Predicted', (predicted_center[0]+10, predicted_center[1]), font, fontsize, self.Green, font_thickness, cv2.LINE_AA)
+        #cv2.putText(frame,'Center', (center[0]-60, center[1]), font, fontsize, self.Red, font_thickness, cv2.LINE_AA)
+        cv2.putText(frame,'Estimated', (int(estimated_center[0]-30), int(estimated_center[1]+20)), font, fontsize, self.Blue, font_thickness, cv2.LINE_AA)        
```

Comment:
are the commented out lines in this function supposed to be commented out?

If they're not required please just delete them